lib/core/src/sdk.rs

```diff
@@ -1243,19 +1253,50 @@ impl LiquidSdk {
                 self.pay_liquid(liquid_address_data.clone(), amount_sat, *fees_sat, true)
                     .await
             }
-            SendDestination::Bolt11 { invoice } => {
-                self.pay_bolt11_invoice(&invoice.bolt11, *fees_sat).await
+            SendDestination::Bolt11 {
+                invoice,
+                bip353_address,
+            } => {
+                let response = self.pay_bolt11_invoice(&invoice.bolt11, *fees_sat).await?;
+                if let (Some(tx_id), Some(destination)) =
+                    (&response.payment.tx_id, &response.payment.destination)
+                {
+                    self.persister
+                        .insert_or_update_payment_details(PaymentTxDetails {
+                            tx_id: tx_id.clone(),
+                            destination: destination.clone(),
+                            description: None,
+                            lnurl_info: None,
+                            bip353_address: bip353_address.clone(),
+                        })?;
+                }
```

Comment:
I think there's no point inserting if the bip353_address is not set, same for the bolt12